spot_driver/spot_driver/spot_ros2.py

```diff
@@ -495,6 +497,10 @@ def __init__(self, parameter_list: Optional[typing.List[Parameter]] = None, **kw
 
         self.create_subscription(Twist, "cmd_vel", self.cmd_velocity_callback, 1, callback_group=self.group)
         self.create_subscription(Pose, "body_pose", self.body_pose_callback, 1, callback_group=self.group)
+        if has_arm:
+            self.create_subscription(
+                ArmVelocityCommandRequest, "arm_velocity", self.set_arm_velocity_callback, 1, callback_group=self.group
+            )
```

Comment:
+1, to keep with the style of the rest of the repo I think it would be best to handle this as either a service or an action. We have methods for sending standard robot commands as either an action (https://github.com/bdaiinstitute/spot_ros2/blob/492587a8962c5796994b633bb451c5e1a86c8906/spot_driver/spot_driver/spot_ros2.py#L2231) or a service (https://github.com/bdaiinstitute/spot_ros2/blob/492587a8962c5796994b633bb451c5e1a86c8906/spot_driver/spot_driver/spot_ros2.py#L2216) -- we use the action implementation for longer commands and the service implementation for faster/"streaming" commands. If you want more direct control you should be able to achieve the same behavior you've written here with a service call

Reply:
Hey normally I would agree to keep the whole repo ins the same style but in this case I had specific reasons to disregard that and implement it another way.
The ArmVelocity movement is more related to basic movemnet of the robot or in this case the cmd_vel interface.
Using a service at this point would have too much time overhead which would result in way less smooth movement ( because, yes i first tried it as a service and it was horribly unresponsive). Even though services are kind of fast they still require all the internal handshakes and are not suitable for this kind of control.

Reply:
We discussed this some more and it sounds good to keep it as a subscription to be analogous to `cmd_vel` but for the arm